File: Coding skills/parityDegree.cpp
// The problem: https://app.codility.com/programmers/trainings/5/parity_degree/
#include <iostream>
#include <bits/stdc++.h>
#include <algorithm>
using namespace std;

int highest(int N)
{
    int maxPower = log2(N);
    for (int i = maxPower; i > 0; i--)
    {
        int factor = pow(2, i);
        int mod = N % factor;
        if (mod == 0)
        {
            return i;
        }
    }
    return 0;
};

int main()
{
    cout << highest(0) << endl;
    return 0;
};

File: Coding skills/parity_degree.py
# The problem: https://app.codility.com/programmers/trainings/5/parity_degree/

def highest_power(N):
    # the binary representation without the 'b'
    binar = bin(N)[2:].zfill(8)
    # convert it to a list
    lst = [int(x) for x in str(binar)]
    # reverse the list
    lst.reverse()
    # find the first 'one' from the right
    for i in range(len(lst)):
        if lst[i] == 1:
            return i
    return 0


if __name__ == '__main__':
    print(highest_power(24))
